pkg/core/interop/context.go

```diff
@@ -45,21 +45,23 @@ type Ledger interface {
 
 // Context represents context in which interops are executed.
 type Context struct {
-	Chain            Ledger
-	Container        hash.Hashable
-	Network          uint32
-	Hardforks        map[string]uint32
-	Natives          []Contract
-	Trigger          trigger.Type
-	Block            *block.Block
-	NonceData        [16]byte
-	Tx               *transaction.Transaction
-	DAO              *dao.Simple
-	Notifications    []state.NotificationEvent
-	Log              *zap.Logger
-	VM               *vm.VM
-	Functions        []Function
-	Invocations      map[util.Uint160]int
+	Chain         Ledger
+	Container     hash.Hashable
+	Network       uint32
+	Hardforks     map[string]uint32
+	Natives       []Contract
+	Trigger       trigger.Type
+	Block         *block.Block
+	NonceData     [16]byte
+	Tx            *transaction.Transaction
+	DAO           *dao.Simple
+	Notifications []state.NotificationEvent
+	Log           *zap.Logger
+	VM            *vm.VM
+	Functions     []Function
+	Invocations   map[util.Uint160]int
+	TxesConsumed  map[util.Uint256]int64
```

Comment:
Not sure it's the most elegant way, but it works for now.

Reply:
Not elegant. I'm still thinking about how to optimize. Any suggestion?

Reply:
Can be made a simple slice of `int64`, because transactions are strictly ordered, a tiny optimization, but still some additional data to pass into the interop context.

Reply:
updated